### DISCUSSION TITLE
Next js deployment with custom server

### DISCUSSION THREAD

**Discussion by KSC2023** [vercel/next.js#61039] [Help]
After running npm start I was getting an error of node evn is not recognised as an internal or external command which i solved using cross evn installation npm install --save-dev cross-env command but after that when I build again and zip file uploaded to cpanel then after creating node js app again problem coming now error code 503 is coming cpanel provide node version 16 max app build in 20 it is my first hello world app trying to test server configuration all answers welcome if solution to this or any other way to achieve the task in cpanel 

Problem resolved by using node js version 20 for that I ask my server provider to update the node on my server thanks to all who help me

**Comment by Nadeera3784** [comment]
Hi @KSC2023 

You encountered a "503 Error" on your cPanel-hosted Node.js app, likely due to server configuration or resource issues; to troubleshoot, ensure your app is compatible with Node.js version 16, review your code for errors, check server logs for clues, optimize resource usage, verify cPanel setup, port configuration, correct ZIP file uploads, and permissions, and consider restarting the app if you make changes.

Check if your app works with Node.js version 16. You can specify this in your package.json like this:

```
"engines": {
  "node": "16.x"
}

```

Ensure that you've configured your cPanel Node.js app correctly, including specifying the entry point file (usually app.js or index.js) and any required environment variables.

**Reply by KSC2023** [reply]
Sorry for the delay in replying I got the point but issue resolved by adding nodejs version 20 to cpanel and redeploy the app thanks for your support and good to see this new approach that was not known to me if I will try this i will surely update about this thanks again hopping for more help as I just started with next js and problems will welcome me for sure 😊

**Reply by Nadeera3784** [reply]
@KSC2023 
Was it a Node version compatibility issue? 
Anyways, I'm glad I could help you.
